src/vs/editor/common/services/resolverService.ts

```diff
@@ -8,23 +8,22 @@ import { TPromise } from 'vs/base/common/winjs.base';
 import URI from 'vs/base/common/uri';
 import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
 import { IModel } from 'vs/editor/common/editorCommon';
-import { ITextEditorModel as IBaseTextEditorModel } from 'vs/platform/editor/common/editor';
-import { IDisposable } from 'vs/base/common/lifecycle';
+import { IEditorModel } from 'vs/platform/editor/common/editor';
+import { IDisposable, IReference } from 'vs/base/common/lifecycle';
 
 export const ITextModelResolverService = createDecorator<ITextModelResolverService>('textModelResolverService');
 
 export interface ITextModelResolverService {
 	_serviceBrand: any;
 
 	/**
-	 * Given a resource, tries to resolve a ITextEditorModel out of it. Will support many schemes like file://, untitled://,
-	 * inMemory:// and for anything else fall back to the model content provider registry.
+	 * Provided a resource URI, it will return a model reference
+	 * which should be disposed once not needed anymore.
 	 */
-	resolve(resource: URI): TPromise<ITextEditorModel>;
+	getModelReference(resource: URI): IReference<TPromise<ITextEditorModel>>;
```

Comment:
@joaomoreno good stuff. the only thing I am uncertain about is the flow of getting the reference and disposing it. Typically we say you need to `dispose` if you called a `create` method. Here we do a `getReference` and then ask clients to `dispose`. 

I think I would also call it `aquire` and `release` in the resolver service so that it becomes clear that you need to dispose. I am also find with `createReference` and `dispose`.